enrolments/tests/test_serializers.py

```diff
@@ -235,32 +218,110 @@ def test_enrolment_serializer(self):
+        )
+        self.guest_field = Field.objects.create(
+            role=Field.GUEST,
+            name="Relationship",
+            question="What's their relationship to your family?",
+            question_type=Field.MULTIPLE_CHOICE,
+            is_default=True,
+            order=1,
+        )
+        self.session = Session.objects.create(
+            season="Spring",
+            year=2021,
+            start_date=date(2021, 5, 15),
+        )
+        self.class_in_session = Class.objects.create(
+            name="Test Class",
+            session_id=self.session.id,
+            facilitator_id=None,
+            attendance=[],
+        )
+        self.family_data = {
+            "email": "weasleys@theorder.com",
+            "cell_number": "123456789",
+            "address": "12 Grimmauld Place",
+            "preferred_comms": "Owl Post",
+        }
+        self.parent_data = {
+            "first_name": "Molly",
+            "last_name": "Weasley",
+            "information": {f"{self.parent_field.id}": "yes"},
+        }
+        self.children_data = [
+            {
+                "first_name": "Ron",
+                "last_name": "Weasley",
+                "information": {f"{self.child_field.id}": "male"},
+            },
+            {
+                "first_name": "Ginny",
+                "last_name": "Weasley",
+                "information": {f"{self.child_field.id}": "female"},
+            },
+        ]
+        self.guests_data = [
+            {
+                "first_name": "Harry",
+                "last_name": "Potter",
+                "information": {f"{self.guest_field.id}": "friend"},
+            }
+        ]
+
+    def test_family_detail_serializer_create(self):
+        family = dict(self.family_data)
+        family["parent"] = self.parent_data
+        family["children"] = self.children_data
+        family["guests"] = self.guests_data
+
+        data = {
+            "family": family,
+            "session": self.session.id,
+            "preferred_class": self.class_in_session.id,
+            "status": Enrolment.REGISTERED,
+        }
+
+        serializer = EnrolmentCreateSerializer(data=data)
         self.assertTrue(serializer.is_valid())
-        serializer.save()
-        self.assertEqual(self.enrolment.session, self.session)
+        enrolment = serializer.save()
 
-    def test_enrolment_update__invalid_class(self):
-        data = dict(self.update_request)
-        data["preferred_class"] = self.class_not_in_session.id
-        serializer = EnrolmentSerializer(self.enrolment, data=data)
-        self.assertFalse(serializer.is_valid())
+        self.assertEqual(
+            enrolment.family.parent.first_name, self.parent_data["first_name"]
+        )
+        self.assertEqual(
+            enrolment.family.parent.last_name, self.parent_data["last_name"]
+        )
+        self.assertEqual(
+            enrolment.family.parent.information, self.parent_data["information"]
+        )
```

Comment:
this current implementation is technically testing behaviour in the FamilyDetailSerializer as well – ideally our unit tests should only test one thing (ie. if the FamilyDetailSerializer is broken, it would fail its own tests, but not the EnrolmentSerializer's tests)

instead of checking that the family was created, maybe we can try:
1. mock the FamilyDetailSerializer's create method and its return value
2. assert that is was called once with certain args (`assert_called_once_with(args)`)
3. assert that the Enrolment was created with the return value of the mocked method

(similar to the [UserCreateSerializer test](https://github.com/uwblueprint/project-read-backend/blob/6c4419e7a42ee907f4f2f765eea3b110874af50f/accounts/tests/serializers/test_user_create.py#L21))

lmk what you think!